Sources/SafeDIMacros/Macros/InstantiableMacro.swift

```diff
@@ -106,20 +106,29 @@ public struct InstantiableMacro: MemberMacro {
                 .contains(where: { $0.isValid(forFulfilling: visitor.dependencies) })
             guard hasMemberwiseInitializerForInjectableProperties else {
                 if visitor.uninitializedNonOptionalPropertyNames.isEmpty {
-                    var initializer = Initializer.generateRequiredInitializer(
-                        for: visitor.dependencies,
-                        declarationType: concreteDeclaration.declType
+                    var membersWithInitializer = declaration.memberBlock.members
+                    membersWithInitializer.insert(
+                        MemberBlockItemSyntax(
+                            leadingTrivia: .newline,
+                            decl: Initializer.generateRequiredInitializer(
+                                for: visitor.dependencies,
+                                declarationType: concreteDeclaration.declType
+                            ),
+                            trailingTrivia: .newline
+                        ),
+                        at: membersWithInitializer.startIndex
                     )
-                    initializer.leadingTrivia = Trivia(stringLiteral: """
-                    // A generated initializer that has one argument per SafeDI-injected property.
-                    // Because this initializer is generated by a Swift Macro, it can not be used by other Swift Macros.
-                    // As a result, this initializer can not be used within a #Preview macro closure.
-                    // This initializer is generated only because you have not written an appropriate initializer yourself.
-                    // Copy/pasting this generated initializer into your code will enable this initializer to be used within other Swift Macros.
-
-                    """)
-                    return [DeclSyntax(initializer)]
-                        + generateForwardedProperties(from: forwardedProperties)
+                    context.diagnose(Diagnostic(
+                        node: Syntax(declaration.memberBlock),
+                        error: FixableInstantiableError.missingRequiredInitializer(.hasOnlyInjectableProperties),
+                        changes: [
+                            .replace(
+                                oldNode: Syntax(declaration.memberBlock.members),
+                                newNode: Syntax(membersWithInitializer)
+                            ),
```

Comment:
@dfed If there is an initializer that fits the pattern of the generated init, you could replace _only_ that initializer with a new initializer rather than adding an initializer into the member block. 

This could alleviate some of the pain of having the initializer not auto-generated with macros.

Reply:
Yup. The issue with doing this is that I only want to emit one and only one error per problem, and I can't be sure which initializer to update if there are many. I can't add multiple fixits to a single error.

Maybe I'm overthinking it and having multiple errors would be okay? But it feels wrong to me.